spot_driver/src/object_sync/object_synchronizer.cpp

```diff
@@ -329,13 +307,18 @@ ObjectSynchronizer::ObjectSynchronizer(const std::shared_ptr<WorldObjectClientIn
       world_object_update_timer_{std::move(world_object_update_timer)},
       tf_broadcaster_timer_{std::move(tf_broadcaster_timer)},
       clock_interface_{std::move(clock_interface)} {
-  const auto spot_name = parameter_interface_->getSpotName();
-  frame_prefix_ = spot_name.empty() ? "" : spot_name + "/";
-
-  preferred_base_frame_ = stripPrefix(parameter_interface_->getTFRoot(), frame_prefix_);
-  preferred_base_frame_with_prefix_ = preferred_base_frame_.find('/') == std::string::npos
-                                          ? frame_prefix_ + preferred_base_frame_
-                                          : preferred_base_frame_;
+  frame_prefix_ = parameter_interface_->getFramePrefixWithDefaultFallback();
+
+  const std::string tf_root = parameter_interface_->getTFRoot();
+  const std::optional<std::string> valid_tf_root =
+      validateFrameWithPrefix(tf_root, frame_prefix_, kValidTFRootFrameNames);
+  preferred_base_frame_ = stripPrefix(valid_tf_root.value_or(kValidTFRootFrameNames[0]), frame_prefix_);
+  preferred_base_frame_with_prefix_ = valid_tf_root.value_or(frame_prefix_ + kValidTFRootFrameNames[0]);
```

Comment:
nit if you consolidate things in `parameter_interface_base.hpp` you can do something like:
```suggestion
  preferred_base_frame_ = stripPrefix(valid_tf_root.value_or(kDefaultTFRoot), frame_prefix_);
  preferred_base_frame_with_prefix_ = valid_tf_root.value_or(frame_prefix_ + kDefaultTFRoot);
```

https://github.com/bdaiinstitute/spot_ros2/blob/702c2fa967735f5cde9d7b16937aea0f408ecbbc/spot_driver/include/spot_driver/interfaces/parameter_interface_base.hpp#L64

Reply:
I believe that this should also be resolved with https://github.com/bdaiinstitute/spot_ros2/pull/506/commits/6247cd96f346a93ecc114ad747121e9d149c2e3e.